tests/e2e/steps/ViewerContext.ts

```diff
@@ -0,0 +1,116 @@
+import { Given, When, Then, DataTable } from '@cucumber/cucumber'
+import { expect } from '@playwright/test'
+import config from '../config'
+
+import { Ocis } from '../pageObjects/Ocis'
+import { Viewer } from '../pageObjects/Viewer'
+import { uploadFile, delay } from '../utils/helpers'
+
+const ocis = new Ocis()
+const viewer = new Viewer()
+
+Given(
+  'the following 3D models have been uploaded:',
+  async function (filesForUpload: DataTable): Promise<void> {
+    for (const file of filesForUpload.hashes()) {
+      await uploadFile(file.filename)
+    }
+  }
+)
+
+Given(
+  'the user has logged in with username {string} and password {string}',
+  async function (user: string, password: string): Promise<void> {
+    await global.page.goto(config.baseUrlOcis)
+    await ocis.login({ username: user, password: password })
+  }
+)
+
+When(
+  'the user previews the file {string} in the 3D model viewer',
+  async function (filename: string): Promise<void> {
+    await ocis.previewFile(filename)
+  }
+)
+
+Then(
+  'the 3D model {string} should be displayed in the viewport',
+  async function (filename: string): Promise<void> {
+    // check if viewport and canvas are visible
+    await expect(global.page.locator(viewer.elements.modelViewport)).toBeVisible()
+    await expect(global.page.locator(viewer.elements.modelViewportCanvas)).toBeVisible()
+    // add some delay to allow model to be loaded
+    await delay(1000)
```

Comment:
would be great if we could have more smart ways to wait instead of static wait.
```suggestion
```

Reply:
@saw-jan replaced it with page.waitForSelector() - is that ok or is there a better approach?

Reply:
`page.waitForSelector()` is better